spark-jobs/src/main/scala/za/co/absa/enceladus/common/SparkCompatibility.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.common
+
+/**
+ * This is where one can set which versions of Spark are supported. Shared between
+ * [[za.co.absa.enceladus.standardization.StandardizationJob]] and [[za.co.absa.enceladus.conformance.DynamicConformanceJob]]
+ *
+ * Expected to be used by [[version.SparkVersionGuard]]
+ */
+
+import za.co.absa.commons.version.Version._
+import za.co.absa.commons.version.impl.SemVer20Impl.SemanticVersion
+
+object SparkCompatibility {
+  val minSparkVersionIncluded: SemanticVersion = semver"2.4.2"
+  val maxSparkVersionExcluded: SemanticVersion = semver"3.0.0"
```

Comment:
Why to limit to max version? Do we already know about some incompatibility?
With the max version the SW can age faster then necessary.

Reply:
I see your point, @yruslan's request was to limit at <3.0.0, which I did.
![image](https://user-images.githubusercontent.com/4457378/77650335-a514a980-6f6b-11ea-84a4-0765cc028a5b.png)

 @yruslan, should we have the option to disable the upper bound then?
I can certainly do it (as @benedeki suggests - make the "max" optional in the `SparkVersionGuard`, but pass defined "max" value for our spark jobs...?

Reply:
As suggested, I have adjusted the `SparkVersionGuard` so that the upper bound is now optional. The spark-jobs currently still enforce `3.0.0` for the upper bound, but it would now be very easy to remove, should we decide to do so. 😊

Reply:
Chatted with @yruslan and concluded, that a good compromise would be too issue a warning in case the version is over max. Not to prevent run altogether.

Reply:
Hey guys, sorry for replying late. Will try to be more responsive.

Reply:
Done - there is a `log.warn` now for the case where spark version exceeds the version upper bound.